### DISCUSSION TITLE
Can I have a project with no port exposed?

### DISCUSSION THREAD

**Discussion by Paillat-dev** [coollabsio/coolify#3016] [Q&A]
I'm new to Coolify and similar systems, and I have multiple applications I'd like to run on this platform. My main question is: Am I required to expose a port for my app and link it to a website? 

I have different types of applications, inlcuding:
1. Discord bots
2. Recurring tasks

None of these expose any ports. Is it still possible to add them to Coolify?

**Comment by nuxwin** [comment]
Good morning,

No, it is not mandatory at all. For instance, you can deploy a docker services stack using a docker-compose configuration, which define one to many services that don't expose ports.

For instance, see the screenshot below, you can see many services. For each services, you can enter one to many domains (with different mapping). However, if you don't set any domains, no Traefik configuration will be generated and so, your services will not be exposed to the WAN via traefik. This is exactly what you want.


![Capture d’écran 2024-08-07 à 16 00 28](https://github.com/user-attachments/assets/f340a90a-55e2-4604-8f69-8ace8f4a7562)
![Capture d’écran 2024-08-07 à 16 00 17](https://github.com/user-attachments/assets/9734843e-4091-4750-8ba6-b349d6a01521)

**Reply by Paillat-dev** [reply]
Thank you very much, I did not understand that I could omit domain to disable it !

**Reply by nuxwin** [reply]
You"re welcome.

**Reply by Paillat-dev** [reply]
Actually, it still asks for a port, which seems wrong...

**Reply by mmoreiras** [reply]
same error here, it still asking for a port
